### PR TITLE
Fix license headers in all source files

### PR DESCRIPTION
**Pull request by tmat**
Applies code fix to update source file headers.
Some files had different wording, some incorrectly referred to Apache license, some had Xamarin copyright, some did not have any license header.

### CODE REVIEW

**Comment by tmat** [comment]
@richlander

**Review by richlander — APPROVED** [review]

**Comment by richlander** [comment]
Thanks for the fixes.

File: .editorconfig [hunk 1]
```diff
@@ -161,7 +161,7 @@ csharp_space_between_square_brackets = false
 dotnet_code_quality.api_surface = all
 
 # License header
-file_header_template = Copyright (c) .NET Foundation and contributors. All rights reserved.\nLicensed under the MIT license. See LICENSE file in the project root for full license information.\n
+file_header_template = Copyright (c) .NET Foundation and contributors. All rights reserved.\nLicensed under the MIT license. See LICENSE file in the project root for full license information.
```
Comment (mavasani) [inline .editorconfig:164]: I see that the diagnostic is marked as a warning: https://github.com/dotnet/sdk/blob/dfae9b7b85f2b693a9a55f8d70b08af01d3ee83f/.editorconfig#L335-L336

Why isn't this enforced on CI? Or are none of the code style rules enforced on build on this repo?

Comment (tmat) [reply .editorconfig:164]: Not sure. Roslyn's model of running a separate "correctness" CI leg would be nice, but I don't think this repo has it.

File: src/ApiCompat/Microsoft.DotNet.ApiCompatibility/Logging/ISuppressionEngine.cs [hunk 2]
```diff
@@ -1,5 +1,5 @@
-﻿// Licensed to the .NET Foundation under one or more agreements.
-// The .NET Foundation licenses this file to you under the MIT license.
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
```
Comment (ViktorHofer) [inline src/ApiCompat/Microsoft.DotNet.ApiCompatibility/Logging/ISuppressionEngine.cs:2]: Why are making this change? dotnet/runtime uses the existing license header everywhere: https://github.com/dotnet/runtime/blob/5d370f31305012b074f070337ace39ca1b5aedeb/src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.cs#L1-L2

cc @jkotas

Comment (jkotas) [reply src/ApiCompat/Microsoft.DotNet.ApiCompatibility/Logging/ISuppressionEngine.cs:2]: "Licensed to the .NET Foundation under one or more agreements." is the correct header to use in .NET Foundation projects.

See https://github.com/dotnet-foundation/foundation/blob/master/guidance/copyright.md#file-headers . 

dotnet/runtime further shortened it to 2 lines, both the make it shorter and to address some problems with the 3 line version. See https://github.com/dotnet/runtime/pull/38732 for rationale.

File: src/Cli/Microsoft.DotNet.Cli.Sln.Internal/FileManipulation/FileUtil.cs [hunk 3]
```diff
@@ -1,28 +1,5 @@
-//
-// FileUtil.cs
-//
-// Author:
-//       Lluis Sanchez Gual <lluis@xamarin.com>
-//
-// Copyright (c) 2015 Xamarin, Inc (http://www.xamarin.com)
```
Comment (jkotas) [inline src/Cli/Microsoft.DotNet.Cli.Sln.Internal/FileManipulation/FileUtil.cs:7]: In general, it is not ok to delete Xamarin and Novell copyrights like this (in dotnet/runtime at least).

File: src/ApiCompat/Microsoft.DotNet.ApiCompatibility/Logging/ISuppressionEngine.cs (hunk 2 above)
Comment (tmat) [reply src/ApiCompat/Microsoft.DotNet.ApiCompatibility/Logging/ISuppressionEngine.cs:2]: Sounds good. I'll update the editorconfig based on dotnet/runtime repo and run the quick fix again.

File: src/Cli/Microsoft.DotNet.Cli.Sln.Internal/FileManipulation/FileUtil.cs (hunk 3 above)
Comment (tmat) [reply src/Cli/Microsoft.DotNet.Cli.Sln.Internal/FileManipulation/FileUtil.cs]: Done.

**Review by jkotas — APPROVED** [review]

**Pull request merged**